platform/http_client_apple.mm

```diff
@@ -58,10 +103,12 @@ - (NSData *)sendSynchronousRequest:(NSURLRequest *)request
 
   dispatch_group_t group = dispatch_group_create();
   dispatch_group_enter(group);
+  
+  RedirectDelegate * delegate = [[RedirectDelegate alloc] init: handleRedirects];
 
   [[[HttpSessionManager sharedManager]
       dataTaskWithRequest:request
-                 delegate:nil
```

Comment:
Can `self` implement the delegate in a simpler way?

Reply:
Meaning: will it be simpler if instead of passing an external delegate, it will be implemented in this class, and `delegate:self` will be passed to the function?